### PR TITLE
No more comment spam

### PR DESCRIPTION
**Pull request by aduffeck**
We recently ran into the problem that we needed to run two prophets on our project: one for unit tests, which should report the results quickly and one for the integration tests which take quite some time to complete.

That was not possible until now, though, because prophet uses comments to track the source and target SHAs and adding another set of comment users and doubling the comment count on the pull requests was not an option. We also wanted to disable the commenting completely which also wasn't possible.

This pull request now changes prophet to use the status to keep track of things instead of comments. Since one commit can have multiple statuses which are distinguished by the context we can now have two prophets (with context 'prophet/unit' and 'prophet/integration') and finally get rid of the comments.

### CODE REVIEW

**Comment by b4mboo** [comment]
Overall, this approach is way better. At the time Prophet has been written GitHub didn't support multiple statuses yet (those bastards :trollface: ). Thanks for all your effort. Once we find a way to manually trigger Prophet runs without the comments, this is probably ready to merge. :+1:

File: spec/prophet_spec.rb [hunk 1]
```diff
@@ -31,19 +31,20 @@
   end
 
   it 'loops through all open pull requests' do
-    @github.should_receive(:pulls).with(@project, 'open').and_return([@api_response])
+    @github.should_receive(:pulls).with(@project, {state: "open"}).and_return([@api_response])
```
Comment (b4mboo) [inline spec/prophet_spec.rb]: Unnecessary double quotes are a show stopper for me. :trollface: :laughing:

Comment (b4mboo) [inline spec/prophet_spec.rb]: Don't dare to change it back to single quotes now.

Comment (aduffeck) [inline spec/prophet_spec.rb]: Oh how I miss these discussions... ;)

**Comment by aduffeck** [comment]
With the last commit you can still delete the comments to trigger a rerun as before by default, it only doesn't work anymore when comments are disabled. I think this should work for everyone, but we can wait for @digitaltom to actually try it out, there's no need for hurries here.

File: spec/prophet_spec.rb (hunk 1 above)
Comment (b4mboo) [inline spec/prophet_spec.rb]: :100: :+1:

**Comment by b4mboo** [comment]
@digitaltom Once you give me the :+1:, I'll merge it and release a new version of the gem.

**Comment by digitaltom** [comment]
Works fine for us, thanks @aduffeck for switching to the status api and the possibility of multiple prophets. We also can make use of this.

**Comment by b4mboo** [comment]
I still need to release a new version of the gem. Hopefully, this is gonna happen on the weekend.

**Pull request merged**